crates/storage/provider/src/providers/snapshot/writer.rs

```diff
@@ -0,0 +1,250 @@
+                        ),
+                        path,
+                    )
+                }
+                Err(err) => return Err(err),
+            };
+
+        match NippyJarWriter::from_owned(jar) {
+            Ok(writer) => Ok((writer, path)),
+            Err(NippyJarError::FrozenJar) => {
+                // This snapshot has been frozen, so we should
+                Err(ProviderError::FinalizedSnapshot(segment, block))
+            }
+            Err(e) => Err(e.into()),
+        }
+    }
+
+    /// Commits configuration changes to disk and updates the reader index with the new changes.
+    pub fn commit(&mut self) -> ProviderResult<()> {
+        // Commits offsets and new user_header to disk
+        self.writer.commit()?;
+
+        // TODO(joshie): update the index without re-iterating all snapshots
+        self.reader.update_index()?;
+
+        Ok(())
+    }
+
+    /// Allows to increment the [`SegmentHeader`] end block. It will commit the current snapshot,
+    /// and create the next one if we are past the end range.
+    ///
+    /// Returns the current [`BlockNumber`] as seen in the static file.
+    pub fn increment_block(&mut self, segment: SnapshotSegment) -> ProviderResult<BlockNumber> {
+        let last_block = self.writer.user_header().block_end();
+        let writer_range_end = *find_fixed_range(BLOCKS_PER_SNAPSHOT, last_block).end();
+
+        // We have finished the previous snapshot and must freeze it
+        if last_block + 1 > writer_range_end {
+            // Commits offsets and new user_header to disk
+            self.writer.commit()?;
+
+            // Opens the new snapshot
+            let (writer, data_path) = Self::open(segment, last_block + 1, self.reader.clone())?;
+            self.writer = writer;
+            self.data_path = data_path;
+
+            match segment {
+                SnapshotSegment::Headers => todo!(),
+                SnapshotSegment::Transactions | SnapshotSegment::Receipts => {
+                    let block_start =
+                        *find_fixed_range(BLOCKS_PER_SNAPSHOT, last_block + 1).start();
+                    *self.writer.user_header_mut() =
+                        SegmentHeader::new(block_start..=block_start, None, segment)
+                }
+            }
+        } else {
+            match segment {
+                SnapshotSegment::Headers => todo!(),
+                SnapshotSegment::Transactions | SnapshotSegment::Receipts => {
+                    self.writer.user_header_mut().increment_block()
+                }
+            }
+        }
+
+        Ok(self.writer.user_header().block_end())
+    }
+
+    /// Truncates a number of rows from disk. It delets and loads an older snapshot file if block
+    /// goes beyond the start of the current block range.
+    ///
+    /// **last_block** should be passed only with transaction based segments.
+    ///
+    /// ATTENTION: It **requires** `self.commit` to be called manually
```

Comment:
```suggestion
    /// # Note
    /// Requires `commit` to be called manually ...
```

reasoning in place of `...` could be nice

Reply:
also - is this true? if you don't call `commit`, nothing touches the disk?

Reply:
ended up just committing on `truncate`, which is essentially updating the configuration file  on disk